calendar-bundle/src/Resources/contao/classes/Calendar.php

```diff
@@ -638,9 +638,25 @@ public function purgeOldFeeds()
 	 */
 	private function getPageWithDetails($intPageId)
 	{
-		if (!isset(self::$arrPageCache[$intPageId]))
+		if (!\array_key_exists($intPageId, self::$arrPageCache))
 		{
-			self::$arrPageCache[$intPageId] = PageModel::findWithDetails($intPageId);
+			$objPage = self::$arrPageCache[$intPageId] = PageModel::findWithDetails($intPageId);
```

Comment:
I‘m afraid this does not work. `self::$arrPageCache[$intPageId]` will now store the page model **without** the `templateGroup` parameter, which is only set in line 659.

Reply:
`$objPage` is an object reference, thus any changes to `$objPage` will also be present in `self::$arrPageCache[$intPageId]`. I also tested it and it works.